### PR TITLE
Use high-level retry logic

### PR DESCRIPTION
**Pull request by trenton05**
Remove connection error logic which was inaccurate.  Have generic request retry on failure.

### CODE REVIEW

File: src/scripts/bc-api-frame.js [hunk 1]
```diff
@@ -89,35 +89,29 @@ bc.ApiFrame.prototype._receiveApiMessage = function(event) {
...
 			} else {
 				this.messageListener(message.method, message.params, message.id);
 			}
 		} else {
 			var rest = this.framePendingResults[message.id];
 			if(rest) {
 				if(!message.error) {
-					this.framePendingResults[message.id] = null;
+					delete this.framePendingResults[message.id];
 					rest.callback.finished(message);
```
Comment (boldwade) [inline src/scripts/bc-api-frame.js]: Can you change `rest.callback.finished(message)` to `rest.rest.callback.finished(message)`

File: src/scripts/bc-api-frame.js [hunk 2]
```diff
@@ -89,35 +89,29 @@ bc.ApiFrame.prototype._receiveApiMessage = function(event) {
...
 				} else {
-					this.isConnectionError = true;
-					this.frameLoadQueue.push(rest);
 					bc.util.log('** adding to frameLoadQueue : _receiveApiMessage', false, rest);
```
Comment (boldwade) [inline src/scripts/bc-api-frame.js]: And might as well delete this else as it's no longer doing anything.

**Review by boldwade — CHANGES_REQUESTED** [review]

File: src/scripts/bc-api-frame.js (hunk 2 above)
Comment (trenton05) [reply src/scripts/bc-api-frame.js]: It's logging the fact it error'ed, I'll reword

**Comment by boldwade** [comment]
Thanks for you help! This should fix client reconnect issues...

**Pull request merged**